src/libraries/System.IO.Compression/src/System/IO/Compression/DeflateZLib/DeflateStream.cs

```diff
@@ -23,6 +23,7 @@ public partial class DeflateStream : Stream
         private byte[]? _buffer;
         private volatile bool _activeAsyncOperation;
         private bool _wroteBytes;
+        private bool _usingGZip;
```

Comment:
We should not special case gzip. Instead, we should change the behavior for all DeflateStream usages, and also make sure the Managed implementation does the same.